### PR TITLE
Webscale Logging

### PR DESCRIPTION
**Pull request by jb3**
Adds a Loki installation to make our logging webscale.

Log querying is:
- Faster (through replicas and caching)
- Easier (through new Grafana plugins and features of Loki 3.0)
- WEBSCALE (through the % increase of Golang in our stack)

### CODE REVIEW

**Comment by jb3** [comment]
Converting this to a draft temporarily to address the following:

- [ ] Update Olli to factor in change of API URL
- [ ] Investigate potential benefits of hooking Loki into AlertManager (could be nothing, but there is an option and I'm not sure what it does)
- [ ] Add Prometheus scrape flags onto Loki as well as the Loki Canaries to monitor log aggregation issues
  - [ ] Related to this, seeing if Minio has a Prometheus endpoint and scraping off that might not be a bad shout.

**Comment by MarkKoz** [comment]
Hi Joe, thanks for your hard work on this draft. Product would like some more data on this "webscale" concept. Can we quantify by how much our service has become more webscale and what the customer impact is? 

We've recently been having robust conversations with Bingus Ltd, one of our largest prospective enterprise customers, about their concerns migrating to our service. They expressed doubt over whether our service is sufficiently webscale to handle the kind of traffic they expect to bring our way. Product is seriously considering adding webscale to our 2024 roadmap. Looking forward to hearing about how we can WOW our customers with this technology! - M

File: kubernetes/namespaces/default/grafana/deployment-grafana.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ spec:
     spec:
       containers:
         - name: grafana
-          image: grafana/grafana:10.4.1
+          image: grafana/grafana:11.0.0-preview
```
Comment (MarkKoz) [inline kubernetes/namespaces/default/grafana/deployment-grafana.yaml:19]: Cutting-edge. LOVE it!

File: kubernetes/namespaces/loki/loki_values.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+read:
+  replicas: 0
+write:
+  replicas: 0
+backend:
+  replicas: 0
+
+# Enable Minio for S3 storage
+minio:
+  enabled: true
```
Comment (MarkKoz) [inline kubernetes/namespaces/loki/loki_values.yml:64]: How does this magic work? Are we deploying minio ourselves? How does loki know where it is?

**Comment by jb3** [comment]
> Hi Joe, thanks for your hard work on this draft. Product would like some more data on this "webscale" concept. Can we quantify by how much our service has become more webscale and what the customer impact is?

Mark, THANK YOU!

We now deploy 3 REPLICATED versions of Loki, all able to ingest and process logs, we load balance the distribution of logs to AGENTS (NOT government ones), located on each node.

We deploy CANARIES to ensure that Loki is active and able to ingest logs.

We deploy MINIO to retain our distributed logs, it is AI-READY!

We deploy a GATEWAY to bring traffic to one of the 3 Loki replicas.

We deploy a CACHE to cache chunks of logs for queries without having to reach out to Minio, this uses [MEMCACHED](https://youtu.be/b2F-DItXtZs?t=215)! (WEBSCALE, SOFTWARE!)

File: kubernetes/namespaces/default/grafana/deployment-grafana.yaml (hunk 1 above)
Comment (jb3) [reply kubernetes/namespaces/default/grafana/deployment-grafana.yaml:19]: THANK YOU! Mark!

**Review by ChrisLovering — APPROVED** [review]

**Pull request merged**